pkg/chains/storage/oci/legacy.go

```diff
@@ -279,15 +280,29 @@ func (b *Backend) RetrieveArtifact(ctx context.Context, obj objects.TektonObject
 }
 
 func newDigest(cfg config.Config, imageName string) (name.Digest, error) {
```

Comment:
I understand the problem now. So I think this needs a bit more work. The result of `newDigest` is used in two different places. Once to configure the storage, and once to configure the artifact being signed.

We want to take into account `storage.oci.repository` only for the storage piece of it. In other words, we don't want to create a signature/attestation with `storage.oci.repository` as its subject.

This appears to have broken in https://github.com/tektoncd/chains/pull/842.

I suggest the following:

1. Introduce a new helper function, e.g. newRepo, which returns the repo which is fed into `WithTargetRepository`. This should use `storage.oci.repository` if set. Otherwise, just use the image's repo.
2. Change newDigest to just parse the given image ref without making any modifications. Its return values should be used as the `Artifact` attribute in the `StoreRequest`.

Reply:
Yeah, make sense, I have updated the pr accordingly